src/contex/data-contex.js

```diff
@@ -0,0 +1,60 @@
+import { useContext, createContext, useEffect, useReducer } from "react";
+import axios from "axios";
+import { DataReducer } from "../reducer/DataReducer";
+
+const ProductData = createContext();
+
+const ProductDataProvider = ({ children }) => {
+  const initialState = {
+    products: [],
+    categories: [],
+    priceRange: "",
+    sortBy: "",
+    filters: {
+      filterByCategories: [],
+      includeOutOfStock: false,
+      fastDelivery: false,
+    },
+  };
+
+  const [state, dispatch] = useReducer(DataReducer, initialState);
+
+  const fetchData = async () => {
+    try {
+      await axios.get("/api/products").then((response) => {
+        dispatch({
+          type: "GET_PRODUCTS",
+          payload: response.data.products,
+        });
+      });
```

Comment:
The whole point of using `async await` was to avoid using `then`, it is used only in Promises, avoid using then. You should return it like this:-
const response = await axios.get("/api/products");
And then you can check response status and do whatever you want to do.

Reply:
yes this make sense i will replace it, thanks for pointing out